pkg/utils/resources/resources.go

```diff
@@ -93,20 +95,58 @@ func Subtract(lhs, rhs v1.ResourceList) v1.ResourceList {
 	return result
 }
 
-// Ceiling calculates the max between the sum of container resources and max of initContainers
+// inspired from https://github.com/kubernetes/kubernetes/blob/e2afa175e4077d767745246662170acd86affeaf/pkg/api/v1/resource/helpers.go#L96
+// https://kubernetes.io/blog/2023/08/25/native-sidecar-containers/
+// Ceiling calculates the max between the sum of container resources and max of initContainers along with sidecar feature consideration
 func Ceiling(pod *v1.Pod) v1.ResourceRequirements {
 	var resources v1.ResourceRequirements
+
 	for _, container := range pod.Spec.Containers {
 		resources.Requests = MergeInto(resources.Requests, MergeResourceLimitsIntoRequests(container))
 		resources.Limits = MergeInto(resources.Limits, container.Resources.Limits)
 	}
+
+	restartableInitContainerReqs := v1.ResourceList{}
+	maxInitContainerReqs := v1.ResourceList{}
+	restartableInitContainerLimits := v1.ResourceList{}
+	maxInitContainerLimits := v1.ResourceList{}
+
 	for _, container := range pod.Spec.InitContainers {
-		resources.Requests = MaxResources(resources.Requests, MergeResourceLimitsIntoRequests(container))
-		resources.Limits = MaxResources(resources.Limits, container.Resources.Limits)
+		containerReqs := MergeResourceLimitsIntoRequests(container)
+		containerLimits := container.Resources.Limits
+
+		if container.RestartPolicy != nil && lo.FromPtr(container.RestartPolicy) == v1.ContainerRestartPolicyAlways {
```

Comment:
```suggestion
		if lo.FromPtr(container.RestartPolicy) == v1.ContainerRestartPolicyAlways {
```

If using `lo.FromPtr`, you don't need to check for `nil`